### PR TITLE
Component attribute upsert

### PR DESCRIPTION
**Pull request by joelrebel**
This simplifies work for clients when updating component information,
by performing an update when a certain attribute is present OR an
insert when it's not already present.

Without this change, clients get returned a 400 - 'no sql rows in result',
when a component attribute isn't present, which is then cumbersome to deal with on the clientside.

Note:

The update, insert could have been swapped with the sqlboil `Upsert()` method,
although since the attributes table contains a partial index - "WHERE server_component_id is not null"
and the sqlboil `Upsert()` method has no way of specifying query mods,
the current `Upsert()` method does not work for this case (the matching row for update is not found in the row scan).

https://github.com/volatiletech/sqlboiler/issues/856

### CODE REVIEW

File: pkg/api/v1/router_server_components.go [hunk 1]
```diff
@@ -339,39 +339,56 @@ func (r *Router) serverComponentUpdate(c *gin.Context) {
...
+			// although since the attributes table contains a partial index - "WHERE server_component_id is not null"
+			// and the sqlboil Upsert() method has no way of specifying query mods,
+			// the current Upsert() method does not work for this case (the matching row for update is not found in the row scan).
+			//
+			// https://github.com/volatiletech/sqlboiler/issues/856
+			//
+
+			// update attribute when an attribute matching server_component_id, namespace was found
 			match, err := models.Attributes(
 				qm.Where("server_component_id=?", dbSrvComponent.ID),
```
Comment (sfunkhouser) [inline pkg/api/v1/router_server_components.go:357]: Should this be `dbAttributes.ServerComponentID` instead of `dbSrvComponent.ID`?

Comment (joelrebel) [reply pkg/api/v1/router_server_components.go:357]: `dbAttributes.ServerComponentID` is being assigned the value from `dbSrvComponent.ID` in https://github.com/metal-toolbox/hollow-serverservice/blob/d57f459af96e4f30252f1d511ce71d6c9a8e6951/pkg/api/v1/router_server_components.go#L342 so they are they same here

Comment (sfunkhouser) [reply pkg/api/v1/router_server_components.go:357]: Yeah, I know. It just seemed weird that you assigned this on the line above and then didn't use it

**Review by sfunkhouser — APPROVED** [review]

File: pkg/api/v1/router_server_components.go (hunk 1 above)
Comment (joelrebel) [reply pkg/api/v1/router_server_components.go:357]: oh right, yeah its being assigned to `dbAttributes` for the insert, which is further below.

**Pull request merged**